UOP1_Project/Assets/Scripts/StateMachine/Base/StateNode.cs

```diff
@@ -0,0 +1,120 @@
+﻿
+using System;
+using UnityEngine;
+
+namespace AV.Logic
+{
+    [CreateAssetMenu(menuName = "State Machine/State Node")]
+    public class StateNode : ScriptableState
+    {
+        [Serializable]
+        public struct ActionUsage
+        {
+            public StateTrigger trigger;
+            public StateAction logic;
+        }
+        public ActionUsage[] actions;
+        public StateTransition[] transitions;
+
+        internal void UpdateActions(StateMachine machine)
+        {
+            foreach (var action in actions)
+            {
+                if (action.trigger != StateTrigger.OnUpdate)
+                    continue;
+
+                UpdateAction(action);
+            }
+        }
+
+        private void UpdateAction(ActionUsage action)
+        {
+            if (!action.logic)
+            {
+                Debug.Log($"{name} (StateNode) has unassigned action.", this);
+                return;
+            }
+                
+            action.logic.BeginUpdate(machine);
+        }
+
+        internal void CheckTransitions(StateMachine machine)
+        {
+            // Prioritised transition index
+            // [0] transitions have highest priority
+            // Example:
+            // We see player and start chase, but if we hungry and see food - we chase food instead
+            // If watching for player is higher on the list, we'll not care about food while chasing player
+            // TODO: Make priority matter
+            var priority = int.MaxValue;
+            
+            for (var i = 0; i < transitions.Length; i++)
```

Comment:
I might suggest splitting chunks of this method into smaller private methods for readability, it's a bit large. It'd be really easy at that point to see that this chunk achieves 'X' while the other chunk achieves 'Y' without spending too much time scanning over the whole method 😄